src/base.ts

```diff
@@ -29,6 +33,91 @@ export abstract class Base {
     this.baseUrl = "http://localhost:3000";
     this.disabledDataSettings = DisableDataSettings.None;
     this.enabled = true;
+    this.appDetails = {
+      platform_id: null,
+      app_version: null,
+      server_version: null,
+      store_id: null,
+      source_id: null,
+    };
+    this.anonId = this.generateAnonId()
+    this.userDetails = {
+      user_id: this.anonId,
+      email: null,
+      wallet: null
+    }
+  }
+
+  protected generateAnonId(): any {
+    //todo:paul: get anon hash from @christian
+    let hash = 'anon_example123'
```

Comment:
We're going to use a SHA-256 hash which is more secure than sha-1 or md5. 

there are a bunch of sha-256 libraries like crypto-js or jssha.


```
const CryptoJS = require('crypto-js');

const hash = CryptoJS.SHA256("your input string");
console.log(hash.toString(CryptoJS.enc.Hex)); 
```

or 

```
const jsSHA = require('jssha');

const shaObj = new jsSHA("SHA-256", "TEXT");
shaObj.update("your input string");
const hash = shaObj.getHash("HEX");
console.log(hash); // Outputs the SHA-256 hash as a hex string
```